ttnn/cpp/ttnn/operations/data_movement/concat/concat.cpp

```diff
@@ -97,36 +110,60 @@ MassagedConcat build_untilize_rm_retilize_concat(uint8_t queue_id, const MemoryC
             concat_db_print(res, "untilize_rm_retilize required");
             return res;
         },
-        .pre_transform = [](const std::vector<ttnn::Tensor>& tensors, int dim, unsigned int groups) -> OwnedConcatArgs {
+        .pre_transform =
+            [=](const std::vector<ttnn::Tensor>& tensors, int dim, unsigned int groups) -> OwnedConcatArgs {
+            *logical_output_shape = compute_output_shape(tensors, dim);  // store output shape for post-transform
+
             std::vector<ttnn::Tensor> itensors;
             itensors.reserve(tensors.size());
             std::transform(
                 tensors.begin(),
                 tensors.end(),
                 std::back_inserter(itensors),
-                [](const ttnn::Tensor& input_tensor) -> ttnn::Tensor {
+                [=](const ttnn::Tensor& input_tensor) -> ttnn::Tensor {
                     TT_FATAL(
                         input_tensor.get_layout() == ttnn::TILE_LAYOUT,
                         "ttnn.concat: expected all input tensors to be in tile layout");
                     auto untilized_tensor = ttnn::untilize(input_tensor);
-                    // untilized, so now we have a padded rm tensor
+                    // untilized, so now we have a padded rm tensor. we slice to
+                    // remove the padding.
+                    std::vector<uint32_t> begins_vec(input_tensor.get_shape().rank(), 0);
+                    tt::stl::Span<const uint32_t> begins = begins_vec;
+                    tt::stl::Span<const uint32_t> ends = input_tensor.get_logical_shape().view();
+                    std::vector<uint32_t> steps_vec(input_tensor.get_shape().rank(), 1);
+                    tt::stl::Span<const uint32_t> steps = steps_vec;
+
+                    // we now perform a padding-oblivious slice to remove the
+                    // tile padding.
+                    // FIXME: change this to a legit slice call once
+                    // padding-oblivious entry point is uplifted to the slice
+                    // op.
+                    untilized_tensor = operation::run(
+                        SliceDeviceOperation{begins, ends, steps, output_memory_config},
+                        {untilized_tensor},
+                        {},
+                        {std::nullopt},
+                        queue_id)[0];
+
                     untilized_tensor = ttnn::reshape(
```

Comment:
Makes me think we must change reshape to accept logical shape only (not in this PR)

Reply:
yeah, we definitely need to add this to slice; this is a common request from what I've heard since it's been made padding aware.